Automattic/jetpack
# Social: Add support for sharing limits to Classic Editor

#### Changes proposed in this Pull Request:

- Add support for sharing limits in the Classic Editor. 
- Add a notice showing the user how many shares are left.
- When the amount of shares left is smaller than the number of connections, disable all connections by default.
- Make sure the user can't check more connections than shares left.

#### Other information:


#### Jetpack product discussion


N/A

#### Does this pull request change what data or activity we track or use?


No.
#### Testing instructions:






* Install and activate Jetpack Social and the Classic Editor plugin.
* Make sure Jetpack isn't active.
* Enable sharing limits for your site (first line in test plan of D85993-code)
* Add at least two connections to your testing site.
* Start writing a new post.
* Make sure the notice shows with the number of shares left and the upgrade link.
* To test the limit, add `$shares_remaining = 1` to `class-jetpack-social.php:305`.
* Start a new post again, and check the following:
  * All connections are disabled.
  * You can only check one connection at a time.
<img width="273" alt="Screenshot 2022-09-05 at 19 04 49@2x" src="https://user-images.githubusercontent.com/1713699/188493075-047cb137-45c6-404f-8182-91ea41f9168f.png">
<img width="277" alt="image" src="https://user-images.githubusercontent.com/1713699/188487854-54b0fff6-9acb-4f4f-8033-efc04b1d6505.png">
